### ISSUE TITLE
Add new IP to server certificate

### ISSUE THREAD

**Issue by abitrolly** [kubernetes/minikube#1974]
**FEATURE REQUEST**

Control `minikube` on remote host through SSH local port forwarding.

**Environment**:
 **Minikube version** (use `minikube version`): v0.22.1
- **VM Driver** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep DriverName`): kvm
- **ISO version** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep -i ISO` or `minikube ssh cat /etc/VERSION`): testing

**What happened**:

I setup `kubectl` on my host to connect to `127.0.0.5`, set up ssh with port forwarding, but connection fails with:
```
Kubernetes master is running at https://127.0.0.5:8443
I0914 19:57:19.537863   11391 cached_discovery.go:126] skipped caching discovery info due to Get https://127.0.0.5:8443/api: x509: certificate is valid for 192.168.42.242, 10.0.0.1, not 127.0.0.5
```

Where the server certificate is located and how to add `127.0.0.5` to it?

**Comment by greglanthier** [comment]
I happened to be looking into something similar to this over the past weekend.  My use case did not involve ssh port-forwarding but I think what ultimately stopped me is causing the `x509: certificate ...` message @abitrolly reported.

I'll summarise what I learned from the code.  Hopefully this saves someone some time - or at the very least sheds light onto some of the magic that `minikube` weaves under the covers.  Someone who understands the code will need to confirm whether I'm on the right track.  I'm not a contributor.  

(tl;dr: [certs.go#L148](https://github.com/kubernetes/minikube/blob/master/pkg/minikube/bootstrapper/certs.go#L148))

From what I gather `minikube` regenerates the cluster certificates (including those used by the k8s apiserver) each time the process restarts.

This is whats happening when `minikube` writes the message `Setting up certs...` after you run `minikube start`.

```
lappy$ minikube start
Starting local Kubernetes v1.8.0 cluster...
Starting VM...
Getting VM IP address...
Moving files into cluster...
**Setting up certs...**
Connecting to cluster...
Setting up kubeconfig...
Starting cluster components...
Kubectl is now configured to use the cluster.
lappy$
```

Minikube uses `bootstrappers` to manage its local k8s.  There are a couple of bootstrappers supported at the moment: `localkube` and `kubeadm`.  This is important insofar as they both appear to share the same code that regenerates certificates: [pkg/minikube/bootstrapper/certs.go](https://github.com/kubernetes/minikube/blob/master/pkg/minikube/bootstrapper/certs.go).

[Line 148](https://github.com/kubernetes/minikube/blob/master/pkg/minikube/bootstrapper/certs.go#L148) is the interesting bit:

```
  ips:  []net.IP{net.ParseIP(k8s.NodeIP), serviceIP, net.ParseIP("10.0.0.1")},
```

I _think_ this is the list of IPs burned into the k8s apiserver x509 certificate.

If there were some way of including extra IPs in this list that should be enough to have them included in the [subject alternative names](https://en.wikipedia.org/wiki/Subject_Alternative_Name) in the K8s apiserver x509 certificate.

So effectively, in @abitrolly's case we'd want the code to evaluate to something like this:

```
  ips:  []net.IP{"127.0.0.5",net.ParseIP(k8s.NodeIP), serviceIP, net.ParseIP("10.0.0.1")},
```


I'm not an expert by any means but from what I understand of x509 certificates subject alternative names enumerate additional DNS names and IP address a certificate is valid for.
